src/lua/fiber.c

```diff
@@ -667,13 +667,15 @@ static int
 lbox_fiber_cancel(struct lua_State *L)
 {
 	struct fiber *f = lbox_checkfiber(L, 1);
-	fiber_cancel(f);
-	/*
-	 * Check if we're ourselves cancelled.
-	 * This also implements cancel for the case when
-	 * f == fiber().
-	 */
-	luaL_testcancel(L);
+	if (!(f->flags & FIBER_IS_SYSTEM)) {
+		fiber_cancel(f);
+		/*
+		 * Check if we're ourselves cancelled.
+		 * This also implements cancel for the case when
+		 * f == fiber().
+		 */
```

Comment:
Nit: Please always try to keep the diff as small as possible provided, of course, it doesn't hurt code readability - it's good for history. Here, you could apply the following, much less intrusive diff while still keeping the code easy to read:

```diff
diff --git a/src/lua/fiber.c b/src/lua/fiber.c
index ea5bd1a1b209..74b2052641d9 100644
--- a/src/lua/fiber.c
+++ b/src/lua/fiber.c
@@ -667,6 +667,10 @@ static int
 lbox_fiber_cancel(struct lua_State *L)
 {
 	struct fiber *f = lbox_checkfiber(L, 1);
+	if ((f->flags & FIBER_IS_SYSTEM) != 0) {
+		/* System fibers can't be cancelled from Lua. */
+		return 0;
+	}
 	fiber_cancel(f);
 	/*
 	 * Check if we're ourselves cancelled.
```

Reply:
I wanted to make code in 'happy-path' way. AFAIK, it's supposed to work faster due to processor's optimization mechanisms. But I won't insist and fix it.

Reply:
This function is definitely not a hot path so I don't think we should bother. Anyway, if you want to explicitly tell the compiler that one branch is executed more often than another, you should use [`likely()`](https://github.com/tarantool/tarantool/blob/7ec71b4f6b0a15997553529c8e15c2dd415df44f/src/trivia/util.h#L159-L170).